arrows/ocv/image_container.cxx

```diff
@@ -49,23 +49,26 @@ image_container
 ::image_container(const cv::Mat& d, ColorMode cm)
   : data_(d)
 {
-  if(cm != RGB)
+  if(cm != RGB && data_.channels() != 1)
   {
-    if ( data_.channels() == 3 )
+    switch(d.depth())
     {
-      cv::Mat new_color;
-      new_color.create( d.rows, d.cols, d.type() );
-
-      cv::cvtColor(data_, new_color, CV_BGR2RGB);
-      data_ = new_color;
-    }
-    else if ( data_.channels() == 4 )
-    {
-      cv::Mat new_color;
-      new_color.create( d.rows, d.cols, d.type() );
+      case CV_8U:
+      case CV_16U:
+      case CV_32F:
+      {
+        cv::Mat new_color;
+        new_color.create( d.rows, d.cols, d.type() );
 
-      cv::cvtColor(data_, new_color, CV_BGRA2RGBA);
-      data_ = new_color;
+        cv::cvtColor(data_, new_color, (data_.channels() == 3)?CV_BGR2RGB:CV_BGRA2RGBA);
```

Comment:
What if `data_.channels()` is 2 or 5 or more?  It's not a common case, but sometime an image represents things other than color.  The previous code only converted images with 3 or 4 channels.  With this change it will try to convert anything with more than 1 channel.

This actually relates to a bigger issue that should not be addressed in this branch, but we will want to consider later.   Not all 3 channel images are either RGB or BGR.  There are many other color spaces and sometimes the values are not even colors at all, but addressing this issue is a bigger design challenge.